sdk-ext-ui-lib/src/main/java/cash/z/ecc/sdk/ext/ui/ZatoshiExt.kt

```diff
@@ -61,17 +72,20 @@ private fun BigDecimal.convertZecDecimalToFiatDecimal(zecPrice: BigDecimal): Big
     return multiply(zecPrice, MathContext.DECIMAL128)
 }
 
-private fun BigDecimal?.convertFiatDecimalToFiatString(monetarySeparators: MonetarySeparators): String {
-    val decimalFormat = DecimalFormat().apply {
-        isParseBigDecimal = true
+private fun BigDecimal.convertFiatDecimalToFiatString(
+    fiatCurrency: Currency,
+    locale: java.util.Locale,
+    monetarySeparators: MonetarySeparators
+): String {
+    return NumberFormat.getCurrencyInstance(locale).apply {
```

Comment:
In my solution I was inspired by the functions in the SDK, but this solution is better for our needs here. Also removing the currency from strings.xml is a good step.